src/PIL/IcoImagePlugin.py

```diff
@@ -222,8 +229,7 @@ def frame(self, idx: int) -> Image.Image:
             im.tile[0] = d, (0, 0) + im.size, o, a
 
             # figure out where AND mask image starts
-            bpp = header["bpp"]
-            if 32 == bpp:
+            if 32 == header.bpp:
```

Comment:
This sort of defensive programming (["Yoda conditions"](https://en.wikipedia.org/wiki/Yoda_conditions)) can be useful in some languages, but not needed for Python:

```suggestion
            if header.bpp == 32:
```